core/templates/pages/topics-and-skills-dashboard-page/skills-list/skills-list.directive.ts

```diff
@@ -169,7 +169,7 @@ angular.module('oppia').directive('skillsList', [
 
                   $scope.skillSummaries = skillSummaries;
                   $scope.selectedSkillId = '';
-                  $scope.confirm = function() {
+                  $scope.save = function() {
```

Comment:
I'm a little confused about how this is supposed to work. Why does this not call confirm(), whereas the other models do (what's the difference)?

Reply:
Because here we pass the data to the `uibModalInstance.close()` in `save()` itself.
But in other places `confirm()` used to handle the closing.
We can still involve  `confirm()` by not closing the modal directly in `save()`, instead pass the data to `confirm()` and the `confirm() `will handle the closing , as in other places

Reply:
Also @seanlip , we discussed this in this comment [link](https://github.com/oppia/oppia/pull/9341#discussion_r429479364)
Pls let me know if this needs more clarifications

Reply:
I think you can follow whatever is happening in other places, for consistency and to avoid developer confusion. My comment above was more along the lines of: don't introduce a new "pre-confirm hook" since it's not really needed. Does that clarify?

Reply:
Yeah, I see why you've changed these lines @donosco98 

There are two ways of changing the `$scope` name:
- Use `confirm` like before, however, you should call `$uibModalInstance.close` instead of `$scope.confirm`. If you assign `$scope.confirm`, it's going to overwrite the original one from the confirm or cancel modal, so then you should call the original way.
- Use `save` as you're doing here in order to the modal works with HTML.

In my perspective, I think it's ok to use `save` instead of `confirm` because the other way may cause more confusion that this way. Is that ok @seanlip?

Reply:
I would agree with using save(), for consistency. I was going to go with what you suggested initially, which is for save() to call confirm(). Does that sound fine? (Don't overwrite `confirm`.)

Reply:
This sounds good!

Reply:
done